app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/projectManager.ts

```diff
@@ -0,0 +1,239 @@
+/** @file This module defines the Project Manager endpoint. */
+import * as newtype from '../newtype'
+
+// =================
+// === Constants ===
+// =================
+
+/** The address of the socket for the project manager. */
+const PROJECT_MANAGER_ENDPOINT = 'ws://127.0.0.1:30535'
+/** Duration before the {@link ProjectManager} tries to create a WebSocket again. */
+const RETRY_INTERVAL = 1000
+/** Duration after which the {@link ProjectManager} stops re-trying to create a WebSocket. */
+const STOP_TRYING_AFTER = 10000
```

Comment:
constants should either use typed numbers or encode unit in name. If these are milliseconds, these should be named `RETRY_INTERVAL_MS`.